x/local-proxy/proxy/httpproxywrapper/handler.go

```diff
@@ -0,0 +1,108 @@
+package httpproxy
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/Jigsaw-Code/outline-sdk/transport"
+	"github.com/Jigsaw-Code/outline-sdk/x/local-proxy/proxy"
+	"github.com/go-httpproxy/httpproxy"
+)
+
+// NewConnectHandler returns a new ProxyHandler which implements http.Handler interface
+func NewConnectHandler(d transport.StreamDialer, addr ...string) *ProxyHandler {
+	a := ""
+	if len(addr) > 0 {
+		a = addr[0]
+	}
+
+	return &ProxyHandler{
+		Proxy: httpproxy.Proxy{
+			Rt: &http.Transport{
+				TLSClientConfig: &tls.Config{},
+				DialContext:     func(ctx context.Context, network, addr string) (net.Conn, error) { return d.Dial(ctx, addr) },
+			},
+		},
+		address: a,
+	}
+}
+
+// ProxyHandler defines parameters for running an HTTP ProxyHandler. It implements
+// http.Handler interface for ListenAndServe function. If you need, you must
+// set ProxyHandler struct before handling requests.
+type ProxyHandler struct {
+	httpproxy.Proxy
+
+	address string
+	l       net.Listener
+	errC    chan error
+}
+
+var (
+	_ http.Handler = (*ProxyHandler)(nil)
+	_ proxy.Proxy  = (*ProxyHandler)(nil)
+)
+
+func (prx *ProxyHandler) GetAddr() string {
+	return prx.address
+}
+
+func (prx *ProxyHandler) StartServer(addr string) (err error) {
+	// if prx.address is empty listen on random localhost port save it to prx.address and start server
+	if addr != "" {
+		_, _, err := net.SplitHostPort(addr)
+		if err != nil {
+			return fmt.Errorf("invalid address: %w", err)
+		}
+
+		prx.address = addr
+	}
+
+	switch prx.address {
+	case "":
+		prx.l, err = net.Listen("tcp", "localhost:0")
+		if err != nil {
+			return fmt.Errorf("failed to listen on random localhost port: %w", err)
+		}
+
+		prx.address = prx.l.Addr().String()
+	default:
+		prx.l, err = net.Listen("tcp", prx.address)
```

Comment:
Thank you for noticing that.
I thought that it might be a problem if the proxy is started on `:54321` because it'll not be a localhost address (like `localhost:54321`). But it causes problem only in case it's not a localhost address in browser settings, so in fact it's ok.

The only question I have is, maybe we should enforce listening on localhost by default? In another way, a proxy will be available to other devices on the network.